components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSOAPUtils.java

```
@@ -0,0 +1,281 @@
+import javax.xml.soap.SOAPFault;
+import javax.xml.soap.SOAPHeader;
+import javax.xml.soap.SOAPHeaderElement;
+import javax.xml.soap.SOAPMessage;
+import javax.xml.soap.SOAPPart;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+
+/**
+ * A Utility which provides functionality to handle SOAP requests and responses.
+ */
+public class SAMLSOAPUtils {
+
+    private static Log log = LogFactory.getLog(SAMLSOAPUtils.class);
+    private static boolean isBootStrapped = false;
+    public static final String SOAP_FAULT_CODE_CLIENT = "Client";
+    public static final String SAOP_FAULT_CODE_SERVER = "Server";
+
+    /**
+     *
+     * @param authReqStr Authentication Request
+     * @return
+     * @throws IdentityException
+     * @throws IdentitySAML2ECPException
+     */
+    public static XMLObject unmarshall(String authReqStr) throws IdentityException, IdentitySAML2ECPException {
+        InputStream inputStream = null;
+        doBootstrap();
+        try {
+            DocumentBuilderFactory documentBuilderFactory = IdentityUtil.getSecuredDocumentBuilderFactory();
+            DocumentBuilder docBuilder = documentBuilderFactory.newDocumentBuilder();
+            inputStream = new ByteArrayInputStream(authReqStr.trim().getBytes(StandardCharsets.UTF_8));
+            Document document = docBuilder.parse(inputStream);
+            Element element = document.getDocumentElement();
+            UnmarshallerFactory unmarshallerFactory = Configuration.getUnmarshallerFactory();
+            Unmarshaller unmarshaller = unmarshallerFactory.getUnmarshaller(element);
+            return unmarshaller.unmarshall(element);
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            String err = "Error in Processing the  SAML request";
+            throw new IdentitySAML2ECPException(err);
```

Swallowed the original exception.

---

> Swallowed the original exception.

fixed